core/src/main/java/org/apache/calcite/rel/metadata/RelMdRowCount.java

```diff
@@ -83,15 +83,7 @@ public Double getRowCount(RelSubset subset, RelMetadataQuery mq) {
   }
 
   public Double getRowCount(Union rel, RelMetadataQuery mq) {
-    double rowCount = 0.0;
-    for (RelNode input : rel.getInputs()) {
-      Double partialRowCount = mq.getRowCount(input);
-      if (partialRowCount == null) {
-        return null;
-      }
-      rowCount += partialRowCount;
-    }
-    return rowCount;
+    return rel.estimateRowCount(mq);
```

Comment:
`Union.estimateRowCount` doesn't take into account the case that one of the inputs might return null.

Reply:
Thanks @hsyuan , I will fix it.